logstash-core/lib/logstash/java_pipeline.rb

```diff
@@ -378,30 +378,22 @@ def inputworker(plugin)
 
   # initiate the pipeline shutdown sequence
   # this method is intended to be called from outside the pipeline thread
-  # @param before_stop [Proc] code block called before performing stop operation on input plugins
-  def shutdown(&before_stop)
+  def shutdown
```

Comment:
nit: add a note that this is a blocking operation. It was _largely_ a blocking operation before, but the caller _always_ ended up having to `Thread#join` this pipeline's thread too. Now we block on `ShutdownWatcher#start`, which will not exit until either our `Pipeline#thread` is no longer `Thread#alive?` (or a `SystemExit` is thrown).

```suggestion
  # and will block until the pipeline has successfully shut down.
  def shutdown
```